tests/api/api/test_background_tasks.py

```diff
@@ -134,12 +158,139 @@ def test_get_background_task_auth_skip(
+    response = client.get(f"{ORIGINAL_VERSIONED_API_PREFIX}/background-tasks/{name}")
+    assert response.status_code == http.HTTPStatus.OK.value
+    background_task = mlrun.api.schemas.BackgroundTask(**response.json())
+    assert background_task == expected_background_task
+
+
+def test_get_internal_background_task_from_worker_redirect_to_chief_doesnt_exists(
+    db: sqlalchemy.orm.Session, client: fastapi.testclient.TestClient, monkeypatch
+):
+    mlrun.mlconf.httpdb.clusterization.role = "worker"
+    name = "task-name"
+    handler_mock = mlrun.api.utils.clients.chief.Client()
+    handler_mock.get_background_task = unittest.mock.Mock(
+        side_effect=mlrun.errors.MLRunHTTPError()
+    )
+    monkeypatch.setattr(
+        mlrun.api.utils.clients.chief,
+        "Client",
+        lambda *args, **kwargs: handler_mock,
+    )
+    with pytest.raises(mlrun.errors.MLRunHTTPError):
+        client.get(f"{ORIGINAL_VERSIONED_API_PREFIX}/background-tasks/{name}")
+
+
+def test_get_internal_background_task_in_chief_exists(
+    db: sqlalchemy.orm.Session, client: fastapi.testclient.TestClient
+):
+    response = client.post("/test/internal-background-tasks")
+    assert response.status_code == http.HTTPStatus.OK.value
+    background_task = mlrun.api.schemas.BackgroundTask(**response.json())
+    assert background_task.metadata.project is None
+
+    response = client.get(
+        f"{ORIGINAL_VERSIONED_API_PREFIX}/background-tasks/{background_task.metadata.name}"
+    )
+    assert response.status_code == http.HTTPStatus.OK.value
+
+
+def test_trigger_migrations_from_worker_returns_same_response_as_chief(
+    db: sqlalchemy.orm.Session, client: fastapi.testclient.TestClient, monkeypatch
+):
+    mlrun.mlconf.httpdb.clusterization.role = "worker"
+
+    for test_case in [
+        {
+            "status_code": http.HTTPStatus.PRECONDITION_FAILED.value,
+            "content": b'{"error":"Migrations were already triggered and failed. Restart the API to retry"}',
+        },
+        {
+            "status_code": http.HTTPStatus.ACCEPTED.value,
+            "content": b'{"kind":"BackgroundTask","metadata":{"name":"2efd3890-3a12-416d-ae92-807b7796e257",'
+            b'"project":null,"created":"2022-06-13T21:30:42.431158","updated":'
+            b'"2022-06-13T21:30:42.431158","timeout":null},"spec":{},"status":{"state":"running"}}',
+        },
+        {
+            "status_code": http.HTTPStatus.OK.value,
+            "content": b"{}",
+        },
+        {
+            "status_code": http.HTTPStatus.INTERNAL_SERVER_ERROR.value,
+            "content": None,
+        },
+    ]:
+        expected_response = requests.Response()
+        expected_response.status_code = test_case.get("status_code")
+        expected_response._content = test_case.get("content")
+        handler_mock = mlrun.api.utils.clients.chief.Client()
+        handler_mock.trigger_migrations = unittest.mock.Mock(
+            return_value=expected_response
+        )
```

Comment:
While having specific unit tests for each component (the handler, the DB, the chief client, etc...) is good, you also need to make sure you're testing how the components are working together, usually bugs tend to be in the interfaces between components (whether it's a small interface between bt handler to the chief client, or something bigger like ui to api interface)
If you'll look on the code in the chief client, you'll see that when you get a failure status code it's not returning a `requests.Response()` like you're mocking here, it will `mlrun.errors.raise_for_status(response)` and you need to make sure that comes out nice out of the API
Therefore I suggest that the API tests here won't mock the chief client, but will work with a real one that is working against some http server (like in the serving tests, as I suggested initially)

Reply:
I noticed that `to_fail_on_not_ok` thing after writing the above comment, while it seems like the proxying handling is not great but working for the trigger migrations, it seems like it's not good for the get endpoint
And anyways, it only strengthens my case that the testing here is not good enough and you should have end to end tests